src/pybind/mgr/debug/module.py

```diff
@@ -0,0 +1,64 @@
+from mgr_module import CLIReadCommand, HandleCommandResult, MgrModule, Option
+from threading import Event
+from typing import cast, Any, Optional, TYPE_CHECKING
+import errno
+import epdb
+
+def bp(condition: callable = None):
+    if condition:
+        epdb.set_trace_cond(condition)
+    else:
+        epdb.set_trace()
+
+class DebugModule(MgrModule):
+
+    # These are "native" Ceph options that this module cares about.
+    NATIVE_OPTIONS = [
+        'mgr_tick_period',
+    ]
+
+    on = False
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+
+        # set up some members to enable the serve() method and shutdown()
+        self.run = True
+        self.event = Event()
+
+        # ensure config options members are initialized; see config_notify()
+        self.config_notify()
+        self.port = 7777
+
+        self.mgr_tick_period = 5
+
+    @staticmethod
+    def enable():
+        DebugModule.on = True
+
+    @staticmethod
+    def disable():
+        DebugModule.on = False
+        
+    def serve(self) -> None:
+        self.log.info("Starting")
+
+        DebugModule.enable()
+        epdb.serve(port=self.port)
+
+        while self.run:
+            sleep_interval = self.mgr_tick_period
+            self.log.debug('Sleeping for %d seconds', sleep_interval)
+            self.event.wait(sleep_interval)
+            self.event.clear()
+
+    def shutdown(self) -> None:
+        self.log.info('Stopping')
+
+        DebugModule.disable()
+
+        epdb.serve(port=self.port)
```

Comment:
Do we need to call the `epdb.serve` here? it seems not necessary since we are shutting down..

Reply:
oooooooops, I must've copied it there by mistake. At least it is before do_close and do_detach